features/graph/provider/bsm/src/main/java/org/opennms/netmgt/graph/provider/bsm/BusinessServiceGraph.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.graph.provider.bsm;
+
+import org.opennms.netmgt.graph.api.Graph;
+import org.opennms.netmgt.graph.api.generic.GenericEdge;
+import org.opennms.netmgt.graph.api.generic.GenericGraph;
+import org.opennms.netmgt.graph.api.generic.GenericVertex;
+import org.opennms.netmgt.graph.simple.AbstractDomainGraph;
+
+
+// TODO: Patrick: I am not sure we need such a complicated object structure since we only build the objects and then give them
+// as AbstractDomain* to the outside. Evaluate if the builders wouldn't be enough?
```

Comment:
@mvrueden what is your take?

Reply:
This is a dummy implementation of the actual `BusinessServiceGraphProvider`. In order to convert this graph to the `BusinessServiceGraphProvider` which we have to implement at some point we may require a differentiation between types. I'd say do whatever you want here for now, however keep in mind, that we have to properly implement this provider in general.